### PR TITLE
move microwave radar code into library

### PR DESCRIPTION
**Pull request by adkinsjd**
This is part of the effort to abstract functionality of a device (i.e. how the microwave radar works) from the application itself (sending data on the radio, energy adaptivity).

It also makes changes in the way motion is detected. Rather than looking for a threshold it does a sliding window integration (integral [0-window] abs(sample-zero_motion_sample)). With a larger amplification on the hardware, this seems to be working reliably. It can detect motion of just waving from across the room with no false positives.

### CODE REVIEW

File: apps/microwave_radar_module/sense_speed_send_bytes/main.c [hunk 1]
```diff
@@ -265,28 +109,22 @@ int main (void) {
     } while (rc < 0);
     printf(" * Bus initialized\n");
 
+    mr_init();
+
     send_buf[0] = 0x01;
     send_buf[1] = 0x00;
 
     // setup timer
     // set to about two seconds, but a larger prime number so that hopefully we
```
Comment (brghena) [inline apps/microwave_radar_module/sense_speed_send_bytes/main.c]: Fix the comment

File: apps/libsignpost/microwave_radar.c [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    //start ADC sampling at 4khz
+    int ret = adc_set_continuous_sample_callback(adc_callback, NULL);
+    ret |= adc_continuous_sample(MR_ADC_CHANNEL, SAMPLE_FREQ);
+
+    return ret;
+}
+
+//this functionality isn't present in hardware yet
+int mr_disable(void) {
+    return -1;
```
Comment (brghena) [inline apps/libsignpost/microwave_radar.c]: Return `TOCK_ENOSUPPORT` instead

File: apps/libsignpost/microwave_radar.c [hunk 3]
```diff
@@ -0,0 +1,187 @@
+#include <stdint.h>
+#include <stdbool.h>
+#include <tock.h>
+#include <adc.h>
+#include "microwave_radar.h"
+
+#define SAMPLE_FREQ 4000
```
Comment (brghena) [inline apps/libsignpost/microwave_radar.c]: Why 4 kHz? Some comment here explaining the choice would be great.

**Review by brghena — CHANGES_REQUESTED** [review]

**Comment by adkinsjd** [comment]
Okay Branden these changes should be integrated.

**Review by brghena — APPROVED** [review]

**Pull request merged**